AppCenterAnalytics/AppCenterAnalyticsTests/MSDoubleTypedPropertyTests.m

```diff
@@ -0,0 +1,72 @@
+#import "MSDoubleTypedProperty.h"
+#import "MSConstants+Internal.h"
+#import "MSTestFrameworks.h"
+
+@interface MSDoubleTypedPropertyTests : XCTestCase
+
+@end
+
+@implementation MSDoubleTypedPropertyTests
+
+- (void)testNSCodingSerializationAndDeserializationWorks {
+
+  // If
+  MSDoubleTypedProperty *sut = [MSDoubleTypedProperty new];
+  sut.type = @"type";
+  sut.name = @"name";
+  sut.value = 12.23432;
```

Comment:
We should also test with other values using different declaration methods, like `1.23e1`